packages/medusa/src/api/routes/admin/products/update-product.ts

```diff
@@ -186,100 +207,246 @@ import { defaultRelations, defaultFields } from "."
-        sku: Validator.string().allow(null),
-        ean: Validator.string().allow(null),
-        barcode: Validator.string().allow(null),
-        prices: Validator.array().items(
-          Validator.object()
-            .keys({
-              region_id: Validator.string(),
-              currency_code: Validator.string(),
-              amount: Validator.number().integer().required(),
-              sale_amount: Validator.number().optional(),
-            })
-            .xor("region_id", "currency_code")
-        ),
-        options: Validator.array().items({
-          option_id: Validator.string().required(),
-          value: Validator.alternatives(
-            Validator.string(),
-            Validator.number()
-          ).required(),
-        }),
-        inventory_quantity: Validator.number().allow(null),
-        allow_backorder: Validator.boolean().allow(null),
-        manage_inventory: Validator.boolean().allow(null),
-        weight: Validator.number().allow(null).optional(),
-        length: Validator.number().allow(null).optional(),
-        height: Validator.number().allow(null).optional(),
-        width: Validator.number().allow(null).optional(),
-        hs_code: Validator.string().optional().allow(null, ""),
-        origin_country: Validator.string().allow(null, ""),
-        mid_code: Validator.string().allow(null, ""),
-        material: Validator.string().allow(null, ""),
-        metadata: Validator.object().optional(),
-        upc: Validator.string().allow(null),
-      })
-      .optional(),
-    metadata: Validator.object().optional(),
-  })
-
-  const { value, error } = schema.validate(req.body)
-  if (error) {
-    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
-  }
+  const validated = await validator(AdminPostProductsProductReq, req.body)
 
-  const productService = req.scope.resolve("productService")
-  const entityManager = req.scope.resolve("manager")
+  const productService: ProductService = req.scope.resolve("productService")
 
-  await entityManager.transaction(async (manager) => {
-    await productService.withTransaction(manager).update(id, value)
-  })
+  await productService.update(id, validated)
 
   const product = await productService.retrieve(id, {
-    select: defaultFields,
-    relations: defaultRelations,
+    select: defaultAdminProductFields,
+    relations: defaultAdminProductRelations,
   })
 
   res.json({ product })
 }
+
+class ProductTypeReq {
+  @IsString()
+  @IsOptional()
+  id: string
+
+  @IsString()
+  value: string
+}
+
+class ProductTagReq {
+  @IsString()
+  @IsOptional()
+  id: string
+
+  @IsString()
+  value: string
+}
+
+class ProductVariantOptionReq {
+  @IsString()
+  value: string
+
+  @IsString()
+  option_id: string
+}
+
+class ProductVariantPricesReq {
+  @Validate(XorConstraint, ["currency_code"])
+  region_id?: string
+
+  @Validate(XorConstraint, ["region_id"])
+  currency_code?: string
+
+  @IsNumber()
```

Comment:
Again, should be `@IsInteger()`, the same goes for the following `@IsNumber()` (If declared as integers in the oas description)